crates/matrix-sdk/tests/integration/widget.rs

```diff
@@ -310,9 +325,108 @@ async fn read_room_members() {
     }
 }
 
+#[async_test]
+async fn receive_live_events() {
+    let (client, mock_server, driver_handle) = run_test_driver(false).await;
+
+    negotiate_capabilities(
+        &driver_handle,
+        json!([
+            "org.matrix.msc2762.receive.event:m.room.member",
+            "org.matrix.msc2762.receive.event:m.room.message#m.text",
+            "org.matrix.msc2762.receive.state_event:m.room.name#",
+            "org.matrix.msc2762.receive.state_event:m.room.member#@example:localhost",
+        ]),
+    )
+    .await;
```

Comment:
I think it would make more sense to split this, so that we can ensure that e.g. the first capability doesn't imply the third one: right now it's not clear whether an event is filtered by one or the other capability. We could iterate on those capabilities, create a new test driver/client that receives the same sync response below, and check what are the expected received messages for each given capability, instead? (or split in multiple test functions, with some helpers)

Reply:
We already have unit tests for the filters individually, as well as unit tests for the `WidgetMachine`, where we can test edge cases like that better.